source/extensions/quic_listeners/quiche/envoy_quic_utils.h

```diff
@@ -42,8 +42,17 @@ template <class T>
 std::unique_ptr<T> quicHeadersToEnvoyHeaders(const quic::QuicHeaderList& header_list) {
   auto headers = T::create();
   for (const auto& entry : header_list) {
-    // TODO(danzh): Avoid copy by referencing entry as header_list is already validated by QUIC.
-    headers->addCopy(Http::LowerCaseString(entry.first), entry.second);
+    auto key = Http::LowerCaseString(entry.first);
+    if (!Runtime::runtimeFeatureEnabled(
```

Comment:
FWIW I don't think you need to runtime guard QUIC code if youdon't want to as it's still alpha

Reply:
good point! Removed the runtime guard here but still checks it in tests.